### PR TITLE
Added channel_create and channel_delete

### PR DESCRIPTION
**Pull request by ZargorNET**
Just some things I needed and why not contribute it at the same time?

Also bumped version to 0.4.0 according to semver.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ts3-query"
-version = "0.3.0"
+version = "0.4.0"
```
Comment (0xpr03) [inline Cargo.toml]: 0.3.1 as we don't change an existing function (I'm following [semantic](https://semver.org/) versioning with a right shifted "major" pre 1.0)

File: src/data.rs [hunk 2]
```diff
@@ -288,6 +288,118 @@ impl ChannelFull {
...
+    pub channel_name: Option<String>,
+    pub channel_life: Option<ChannelLife>,
+    pub pid: Option<ChannelId>,
+    pub channel_order: Option<ChannelId>,
+    pub channel_topic: Option<String>,
+    pub channel_password: Option<String>,
+    pub channel_maxclients: Option<i32>,
+    pub channel_maxfamilyclients: Option<i32>,
+    pub channel_flag_default: bool,
+    pub channel_codec: Option<i32>,
```
Comment (0xpr03) [inline src/data.rs:311]: Would be nice if we could add a type for this

**Review by 0xpr03 — CHANGES_REQUESTED** [review]
Hey thanks for the PR.
I've added some comments, if possible I'd like to add more documentation (stuff like channel order is kinda confusing what it means). Or for example what the difference between "ChannelEdit" and ChannelFull is. (Struct docs)

Are all of the indentation changes from rustfmt ?

File: src/data.rs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 use crate::raw::*;
-use crate::Result;
+use crate::{Result, Ts3Error};
```
Comment (0xpr03) [inline src/data.rs]: This seems to be unused ?

File: src/data.rs (hunk 2 above)
Comment (ZargorNET) [reply src/data.rs:311]: What do you mean with this? 
I just followed the Channel struct from the [Teamspeak 3 Server Query Manual](https://lethallhost.com.br/clientes/TeamSpeak%203%20Server%20Query%20Manual.pdf)

**Comment by ZargorNET** [comment]
> Are all of the indentation changes from rustfmt ?

I ran `cargo fmt` over it to be sure, but else they're formatted by IntelliJ Rust

**Comment by 0xpr03** [comment]
Yeah please use rustfmt, it's the default for many crates and not IDE specific. (Also it bloated the PR to contain changes at code that isn't related).

**Review by 0xpr03 — APPROVED** [review]
Thanks!

**Comment by 0xpr03** [comment]
Published as 0.3.1

**Pull request merged**